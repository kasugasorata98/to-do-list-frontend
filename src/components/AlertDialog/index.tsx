import {
  AlertDialog as ChakraAlertDialog,
  AlertDialogBody,
  AlertDialogCloseButton,
  AlertDialogContent,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogOverlay,
  Button,
} from "@chakra-ui/react";
import React from "react";

const AlertDialog: React.FC<{
  isOpen: boolean;
  onClose: () => void;
  header?: string;
  description?: string;
  confirm: {
    title?: string;
    onClick: () => void;
  };
  cancel: {
    title?: string;
    onClick: () => void;
  };
}> = ({ isOpen, header, description, confirm, cancel, onClose }) => {
  const cancelRef = React.useRef<any>();

  return (
    <ChakraAlertDialog
      motionPreset="slideInBottom"
      leastDestructiveRef={cancelRef}
      onClose={onClose}
      isOpen={isOpen}
      isCentered
    >
      <AlertDialogOverlay background={"transparent"} />

      <AlertDialogContent>
        <AlertDialogHeader>{header}</AlertDialogHeader>
        <AlertDialogCloseButton />
        <AlertDialogBody>{description}</AlertDialogBody>
        <AlertDialogFooter>
          <Button
            ref={cancelRef}
            onClick={() => {
              onClose();
              cancel.onClick();
            }}
          >
            {cancel?.title}
          </Button>
          <Button colorScheme="red" ml={3} onClick={confirm.onClick}>
            {confirm?.title}
          </Button>
        </AlertDialogFooter>
      </AlertDialogContent>
    </ChakraAlertDialog>
  );
};

export default AlertDialog;
