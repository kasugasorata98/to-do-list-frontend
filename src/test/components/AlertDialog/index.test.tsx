import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import AlertDialog from "@/components/AlertDialog";

describe("AlertDialog component", () => {
  const onClose = jest.fn();
  const onConfirm = jest.fn();
  const onCancel = jest.fn();

  const props = {
    isOpen: true,
    header: "Confirmation",
    description: "Are you sure you want to delete this item?",
    confirm: {
      title: "Delete",
      onClick: onConfirm,
    },
    cancel: {
      title: "Cancel",
      onClick: onCancel,
    },
    onClose,
  };

  it("renders correctly", () => {
    const { asFragment } = render(<AlertDialog {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls onClose function when close button is clicked", () => {
    const { getByLabelText } = render(<AlertDialog {...props} />);
    const closeBtn = getByLabelText("Close");

    fireEvent.click(closeBtn);

    expect(onClose).toHaveBeenCalled();
  });

  it("calls onCancel function when cancel button is clicked", () => {
    const { getByText } = render(<AlertDialog {...props} />);
    const cancelBtn = getByText("Cancel");

    fireEvent.click(cancelBtn);

    expect(onClose).toHaveBeenCalled();
    expect(onCancel).toHaveBeenCalled();
  });

  it("calls onConfirm function when confirm button is clicked", () => {
    const { getByText } = render(<AlertDialog {...props} />);
    const confirmBtn = getByText("Delete");

    fireEvent.click(confirmBtn);

    expect(onConfirm).toHaveBeenCalled();
  });
});
