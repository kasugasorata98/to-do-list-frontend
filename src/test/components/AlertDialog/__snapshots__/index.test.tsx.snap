// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog component renders correctly 1`] = `<DocumentFragment />`;
